### PR TITLE
Extended the hotword detection language to the stt engine and the send message

### PR DESCRIPTION
**Pull request by Joanguitar**
Now the language of the detected hot word is also used for the speech to text engine and is passed to the sent message.
There's also scripts and commands to make it work on Windows.

### CODE REVIEW

**Comment by JarbasAl** [comment]
looks good! small nitpicks before merging

can you made the catalan pocketsphinx model a config option and remove it? if we add one for each language the size of this package gets big really fast, and each model has it's own license

File: setup.py [hunk 1]
```diff
@@ -6,16 +6,23 @@
     packages=['mycroft_voice_satellite', 'mycroft_voice_satellite.speech',
               'mycroft_voice_satellite.res',
               'mycroft_voice_satellite.configuration'],
-    install_requires=["jarbas_hive_mind>=0.10.3",
-                      "speech2text",
+    install_requires=["jarbas_hive_mind",
+                      "SpeechRecognition @ git+https://github.com/Uberi/speech_recognition.git",
```
Comment (JarbasAl) [inline setup.py:10]: why is this needed?

File: setup.py [hunk 2]
```diff
@@ -6,16 +6,23 @@
...
                       "pyee",
                       "requests",
                       "requests_futures",
                       "psutil",
-                      "PyAudio==0.2.11",
-                      "pocketsphinx==0.1.15",
+                      "PyAudio",
+                      "pocketsphinx",
+                      "json_database",
+                      "jarbas_utils",
```
Comment (JarbasAl) [inline setup.py:18]: jarbas_utils has been replaced by ovos_utils

File: setup.py [hunk 3]
```diff
@@ -6,16 +6,23 @@
...
                       "ovos_utils",
-                      "text2speech>=0.1.9"],
+                      "speech2text==0.2.1",
+                      "text2speech @ git+https://github.com/Joanguitar/text2speech.git",
```
Comment (JarbasAl) [inline setup.py:21]: both speech2text and text2speech will be deprecated soon in favor of using https://github.com/OpenVoiceOS/OVOS-plugin-manager

why do you need to install from that branch?

Comment (Joanguitar) [reply setup.py:21]: text2speech is set to work with Linux folders, I added a OS check to replace the temp folder with a Windows one if the OS is Windows.

Comment (Joanguitar) [reply setup.py:21]: I should probably try to PR this repo first, but since you said it will be deprecated I don't see the point, hopefully we can eliminate it soon.

**Comment by JarbasAl** [comment]
please see https://github.com/JarbasHiveMind/HiveMind-voice-sat/pull/19 

it should be merged once @ChanceNCounter approves it, can you rebase on it? it removes the text2speech and speech2text packages, it also uses https://github.com/OpenVoiceOS/ovos-ww-plugin-pocketsphinx which you should be able to point at your model in the config file

File: setup.py (hunk 1 above)
Comment (Joanguitar) [reply setup.py:10]: Removed the git require as it is not needed.

File: setup.py (hunk 2 above)
Comment (Joanguitar) [reply setup.py:18]: Replaced it in the code so only ovos_utils is used

**Comment by Joanguitar** [comment]
I addressed the comments and eliminated both Spanish and Catalan language models.

**Comment by Joanguitar** [comment]
I'm trying to rebase on #19 but it has dependencies to repositories that I are not released yet such as ovos-stt-plugin-chromium>=0.1.1.
I will try to do my best to find working versions of these dependencies, but setup.py won't be able to find these for anyone that's trying to use HiveMin-voice-sat.

**Comment by JarbasAl** [comment]
i think i pushed all missing packages to pip, let me know if i missed some

**Comment by Joanguitar** [comment]
The current version of ovos_utils is not uploaded, therefore ovos_utils.metrics can't be imported.
Second, the configuration used by ovos_utils requires Mycroft to be installed as the default Mycroft configuration is loaded every time a hotword is created through [this](https://github.com/OpenVoiceOS/ovos_utils/blob/dev/ovos_utils/configuration.py#L66) function, which I don't think it should be a requirement for a client terminal.
Finally, ovos_utils configuration is hardcoded to only work on Linux as seen [here](https://github.com/OpenVoiceOS/ovos_utils/blob/dev/ovos_utils/system.py#L13)

**Comment by JarbasAl** [comment]
fixed in latest ovos plugin manager, it should not require the config to exist indeed

**Comment by Joanguitar** [comment]
If we change [this](https://github.com/OpenVoiceOS/ovos-ww-plugin-pocketsphinx/blob/master/ovos_ww_plugin_pocketsphinx/__init__.py#L70):
```
config.set_string('-logfn', '/dev/null')
```
to:
```
logfn = '/dev/null'
if os.name == 'nt': logfn = 'NUL'
config.set_string('-logfn', logfn)
```
the plugin should be compatible with Windows but even after those changes it fails to create Decoder(config).
Later today or tomorrow I will try to find out why.
Also the TTS is failing to be loaded.

**Comment by Joanguitar** [comment]
Sorry, I keep running into more and more problems with #19, I can't find the package with the plugin for Google's TTS a.k.a. "ovos-tts-plugin-google", I found the repository for "ovos-tts-plugin-google-tx" which is a translation plugin and an outdated version in pypi that is indeed recognized as a TTS plugin under the name "ovos_tts_google", but following that outdated package directory I stump across "ovos-tts-plugin-pico".

**Comment by JarbasAl** [comment]
fixed a bug in ovos plugin manager, please be sure to use `ovos-plugin-manager>=0.0.1a12` , should only affect mycroft however not hivemind

the plugins you mentioned are optional, but "ovos-tts-plugin-google-tx" is the correct one indeed, the translate is because that's where the TTS comes from, see the picture in the readme (and the warning!) https://github.com/OpenVoiceOS/ovos-tts-plugin-google-tx

will look into the issue with pico, but that should not matter for this PR

regarding pocketsphinx, the error logs can be super obscure.... what pocketsphinx version do you have installed? might be related to that, at least i have some fuzzy memories about running into that issue previously

**Comment by Joanguitar** [comment]
> 
> 
> fixed in latest ovos plugin manager, it should not require the config to exist indeed

It may be a problem of ovos_utils instead of ovos_plugin_manager:

Traceback (most recent call last):
  File "D:\Git\HiveMind-voice-sat\mycroft_voice_satellite\speech\tts_factory.py", line 72, in create
    tts = clazz(tts_lang, tts_config)
  File "D:\Git\HiveMind-voice-sat\venv\lib\site-packages\ovos_tts_plugin_responsivevoice-0.1.1-py3.9.egg\ovos_tts_plugin_responsivevoice\__init__.py", line 28, in __init__
    super(ResponsiveVoiceTTSPlugin, self).__init__(
  File "D:\Git\HiveMind-voice-sat\venv\lib\site-packages\ovos_plugin_manager\templates\tts.py", line 226, in __init__
    self.spellings = self.load_spellings()
  File "D:\Git\HiveMind-voice-sat\venv\lib\site-packages\ovos_plugin_manager\templates\tts.py", line 239, in load_spellings

    spellings_file = resolve_resource_file(path, config=config)
  File "D:\Git\HiveMind-voice-sat\venv\lib\site-packages\ovos_utils\__init__.py", line 105, in resolve_resource_file
    config = read_mycroft_config()
  File "D:\Git\HiveMind-voice-sat\venv\lib\site-packages\ovos_utils\configuration.py", line 142, in read_mycroft_config
    conf.merge(MycroftDefaultConfig())
  File "D:\Git\HiveMind-voice-sat\venv\lib\site-packages\ovos_utils\configuration.py", line 256, in __init__
    path = find_default_config()
  File "D:\Git\HiveMind-voice-sat\venv\lib\site-packages\ovos_utils\configuration.py", line 68, in find_default_config
    raise FileNotFoundError("Couldn't find mycroft core root folder.")
FileNotFoundError: Couldn't find mycroft core root folder.

**Comment by Joanguitar** [comment]
It is finally working after some small modifications to ovos-plugin-manager, ovos-ww-plugin-pocketsphinx and ovos-tts-plugin-responsivevoice to fix:
-Not depend on Mycroft-core configuration
-Make it compatible with Windows (swapped memory-tempfile with tempfile)
-Corrected the default sample rate in the wake word plugin
-Added language as an option to the TTS
Even though it's working, the log is showing a lot of warnings.

I will squash commits and open PR to those repositories so this branch can be moved to use the OVOS tools.
Should we close this PR until then and continue on top of #19?

**Comment by JarbasAl** [comment]
Thanks for all the work, every PR welcome! I believe i fixed the pocketsphinx issue, the default value was missing a 0

You can close the PR or leave it open and rebase later, whatever is easier for you

**Comment by Joanguitar** [comment]
Cool, I'll rebase it later. I'll try to make some changes in the OVOS plugin to support multiple languages.
